### PR TITLE
Address Point Match

### PR DESCRIPTION
**Pull request by savannahostrowski**
This check identifies Point objects in OSM that have a specified street number (addr:housenumber) 
but no specified street name (addr:street). No specified street name refers to either having a null
value for the street name key, or no street name key present at all.

This pull request includes: all check logic, unit tests, and documentation.

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        // Remove Points that have null as their street name as they cannot be candidates for
+        // a street for the Point of interest
+        final Set<Point> points = pointStream
+                .filter(interiorPoint -> interiorPoint.getTag(POINT_STREET_NAME_KEY).isPresent())
+                .collect(Collectors.toSet());
+
+        // Get all Edges that intersect or are contained within the bounding box
+        final Iterable<Edge> interiorEdges = point.getAtlas().edgesIntersecting(box);
+        // Convert the Iterable into a Stream for filtering
+        final Stream<Edge> edgeStream = StreamSupport.stream(interiorEdges.spliterator(), false);
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java]: Same as above... combine edgeStream & edges into one variable
```java
final Set<Edge> edges = StreamSupport.stream(interiorEdges.spliterator(), false)
   .filter(interiorEdge -> interiorEdge.getTag(EDGE_STREET_NAME_KEY).isPresent())
   .collect(Collectors.toSet());
```

File: src/test/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheckTestRule.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    @TestAtlas(points = {
+            @Point(coordinates = @Loc(value = TEST_1), tags = { "addr:housenumber=20" }),
+            @Point(coordinates = @Loc(value = TEST_2), tags = { "addr:housenumber=20",
+                    "addr:street=Jones" }),
+            @Point(coordinates = @Loc(value = TEST_3), tags = { "addr:housenumber=20",
+                    "addr:street=Jones" }),
+            @Point(coordinates = @Loc(value = TEST_4), tags = { "addr:housenumber=20",
+                    "addr:street=John" }) })
+    private Atlas pointWithStreetNameNoStreetNumberPointCandidatesDuplicateNames;
```
Comment (danielduhh) [inline src/test/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheckTestRule.java]: worlds longest variable name!!! haha, love it ! 😀

**Comment by savannahostrowski** [comment]
@danielduhh Squashed some commits & addressed your comments

**Comment by savannahostrowski** [comment]
Refer to #10

**Pull request closed without merging**